cpp/src/core/cpu_operations.cc

```diff
@@ -21,9 +21,19 @@
 // SOFTWARE.
 
 #include "core/cpu_operations.h"
+#include "Eigen/Dense"  // Dependant on the placement of the library
 
 namespace nice {
+
+// This function creates the transpose of a matrix
 template<typename T>
 Matrix<T> CpuOperations<T>::Transpose(const Matrix<T> &a) {
+  Eigen::a.transposeInPlace();  // In place allows for completion without aliasing effect
```

Comment:
I don't think this will compile because a is not a function or class defined in Eigen. Since a is an Eigen::Matrix, you should be able to directly use a.transposeInPlace()